javascript/ql/src/semmle/javascript/security/dataflow/TaintedPathCustomizations.qll

```diff
@@ -239,6 +240,57 @@ module TaintedPath {
     DataFlow::Node getOutput() { result = output }
   }
 
+  /**
+   * A call that removes all instances of "../" in the prefix of the string.
+   */
+  class DotDotSlashPrefixRemovingReplace extends DataFlow::CallNode {
+    DataFlow::Node input;
+    DataFlow::Node output;
+
+    DotDotSlashPrefixRemovingReplace() {
+      this.getCalleeName() = "replace" and
+      input = getReceiver() and
+      output = this and
+      exists(RegExpLiteral literal, RegExpTerm term |
+        getArgument(0).getALocalSource().asExpr() = literal and
+        (term instanceof RegExpStar or term instanceof RegExpPlus) and
```

Comment:
Minor comment post-merge: this motivates a reuse of `SuperlinearBackTracking::InfiniteRepetitionQuantifier`, but it probably won' matter in this case.

Reply:
This case is not quite the same as `InfiniteRepetitionQuantifier`.   
Here we need the lower-bound to be zero or one occurrences, and `InfiniteRepetitionQuantifier` doesn't care about the lower-bound.  
(And would e.g. flag `/(?:\.\·\/){2,}/`).   

So I think we can skip such a reuse for now.